pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1037,75 +1037,77 @@ public Schema getTableSchema(@Nonnull String tableName) {
    * @throws InvalidTableConfigException
    * @throws TableAlreadyExistsException for offline tables only if the table already exists
    */
-  public void addTable(TableConfig config)
-      throws IOException {
-    TableType type = config.getTableType();
-
+  public void addTable(@Nonnull TableConfig tableConfig) throws IOException {
+    String tableNameWithType = tableConfig.getTableName();
+    TenantConfig tenantConfig;
     if (isSingleTenantCluster()) {
-      TenantConfig tenantConfig = new TenantConfig();
-      tenantConfig.setBroker(
-          ControllerTenantNameBuilder.getBrokerTenantNameForTenant(ControllerTenantNameBuilder.DEFAULT_TENANT_NAME));
-      switch (type) {
-        case OFFLINE:
-          tenantConfig.setServer(ControllerTenantNameBuilder.getOfflineTenantNameForTenant(
-              ControllerTenantNameBuilder.DEFAULT_TENANT_NAME));
-          break;
-        case REALTIME:
-          tenantConfig.setServer(ControllerTenantNameBuilder.getRealtimeTenantNameForTenant(
-              ControllerTenantNameBuilder.DEFAULT_TENANT_NAME));
-          break;
-        default:
-          throw new InvalidTableConfigException("UnSupported table type: " + type);
-      }
-      config.setTenantConfig(tenantConfig);
+      tenantConfig = new TenantConfig();
+      tenantConfig.setBroker(ControllerTenantNameBuilder.DEFAULT_TENANT_NAME);
+      tenantConfig.setServer(ControllerTenantNameBuilder.DEFAULT_TENANT_NAME);
+      tableConfig.setTenantConfig(tenantConfig);
     } else {
-      TenantConfig tenantConfig = config.getTenantConfig();
+      tenantConfig = tableConfig.getTenantConfig();
       if (tenantConfig.getBroker() == null || tenantConfig.getServer() == null) {
-        throw new InvalidTableConfigException("Missing tenant configs");
+        throw new InvalidTableConfigException("Tenant is not configured for table: " + tableNameWithType);
       }
     }
 
-    SegmentsValidationAndRetentionConfig segmentsConfig = config.getValidationConfig();
-    switch (type) {
+    // Check if tenant exists before creating the table
+    TableType tableType = tableConfig.getTableType();
+    String brokerTenantName = ControllerTenantNameBuilder.getBrokerTenantNameForTenant(tenantConfig.getBroker());
+    List<String> brokersForTenant = _helixAdmin.getInstancesInClusterWithTag(_helixClusterName, brokerTenantName);
+    if (brokersForTenant.isEmpty()) {
+      throw new InvalidTableConfigException(
```

Comment:
This exception will be handled on upper level and we will log info/warning there